### ISSUE TITLE
SE10k Internal Server Error

### ISSUE THREAD

**Issue by chrisfirefox** [drobtravels/solaredge-local#20]
I recently got a SE10K installed (I suppose European version, since I am in Austria), and I would very much like to get live information from my inverter. But the only thing it reports back is 
{
    "message": "Internal Server Error"
}
when I call its IP. It is connected via WLAN if that has anything to do with anything. Any idea what that could be? Or where I could ask aside from here? Because my technician definitely used the SetApp process, but in order to do so he directly connected via private WLAN to the inverter, much like a personal hotspot you can create with a smartphone. Maybe that's a new type of security that makes the SetApp interface only work when using the own hotspot?

Any help would be appreciated, thank you!

**Comment by dragoshenron** [comment]
Firmware version?

**Comment by chrisfirefox** [comment]
How can I find out?

**Comment by scheric** [comment]
Go to: https://monitoring.solaredge.com/solaredge-web/p/login 
Login
Go to layout
Click on inverter 
Click on info

**Comment by dragoshenron** [comment]
Yes @scheric .
Or you can activate the local hotspot, connect via setapp and go to commissioning menu -> information

**Comment by chrisfirefox** [comment]
DSP2 Version 2.19.1314
DSP1 Version 1.13.1541
Commcard CPU Version 4.6.27

Anything else you need?

**Comment by scheric** [comment]
@chrisfirefox 
Can you access the inverter commissioning menu trough your browser within your private network.

Is there a possible way that you could hook up an ethernet cable for testing purposes.

**Comment by chrisfirefox** [comment]
Well, not being able to connect to the commissioning menu via browser in my private network is my problem to begin with, instead of showing the commissioning menu every URI I found so far shows the aforementioned {"message":"Internal Server Error"}.

I could easily hook up an ethernet cable if I find an RJ45 port, but tbh I didn't find the port to connect it to so far. The inverter is mounted outside on the wall of my house and I'd really like to not having to open it up one week after installation ;).

**Comment by rogerthn2019** [comment]
I do have a SE7K and I have accessed the commissioning menu as below.
Scan the QR code to get e.g. SSID and password.
Have a WiFi device available and switch shortly the ON/OFF/P switch to P
Connect to the inverter WiFi and browse to 172.16.0.1

**Comment by chrisfirefox** [comment]
The idea behind all this was to have both the commissioning interface via browser and a connection to the internet via my local WLAN so I can continue to use the SolarEdge cloud app AND access real-time data locally. I haven’t tried but am quite sure that your suggestion @rogerthn2019 will most likely work since my contractor used SetApp on his phone to initialize the inverter. Seems to me that 4.6.27 is buggy with the REST interface or that they introduced some kind of handshake or authentication in the app that isn‘t done in the browser, but instead of reporting 401 they just throw an exception which reports as 500...

**Comment by rogerthn2019** [comment]
> The idea behind all this was to have both the commissioning interface via browser and a connection to the internet via my local WLAN so I can continue to use the SolarEdge cloud app AND access real-time data locally. I haven’t tried but am quite sure that your suggestion @rogerthn2019 will most likely work since my contractor used SetApp on his phone to initialize the inverter. Seems to me that 4.6.27 is buggy with the REST interface or that they introduced some kind of handshake or authentication in the app that isn‘t done in the browser, but instead of reporting 401 they just throw an exception which reports as 500...

@chrisfirefox 👍 
**I do agree fully to all of this!**

While waiting for SolarEdge to fix the buggy 4.6.27 let's hope for some sun during the weekend to enable some "checking" using my old Samsung NC10 running Debian stretch, WiFi to the inverter and RJ45 to my local LAN, maybe install Apache and configure with
ProxyPass               "/" "http://172.16.0.1/" and
ProxyPassReverse        "/" "http://172.16.0.1/"

**Comment by scheric** [comment]
You are not the only one that do not have a working API https://github.com/jbuehl/solaredge/issues/124

My inverter API works with Ethernet cable.
CPU-version 4.6.27
DSP1 version 1.0.756
DSP2 version 2.0.704
version WS 1.2.9

**Comment by chrisfirefox** [comment]
Yeah, I just confirmed that when I connect via SetApp ("View information only") and reset the WLAN that I actually get the Web Interface when opening 172.16.0.1 on my phone. So as I suspected (or feared?), they have deactivated commissioning access via WLAN. Doesn't really make sense IMHO because I'm responsible for what enters my WLAN, but that's just me.

Anyways, where would I put the Ethernet cable (didn't find an external Port) and can I connect it to a standard wired network router or is it something special (I think I read RJ485/Modbus) that I need there?

**Comment by dragoshenron** [comment]
you have to open the front panel (DISCONNECT DC AND AC FIRST FOR YOUR SAFETY) and the ethernet port is on the front board, bottom left.

**Comment by scheric** [comment]
@chrisfirefox 
You can use a normal Ethernet cable to attach it to your own network router or switch.
### It is up to you to open up the inverter. 

![solaredge 3 fase pcb](https://user-images.githubusercontent.com/38077357/69549114-f5b5cf00-0f98-11ea-9461-d3a737950ecc.PNG)


If you do open your inverter:
### Be sure that AC is disconnected and Save DC is functioning 
Information for the inverter operating status leds: https://www.solaredge.com/leds 
look at:  System Performance >> Information Inverter is OFF (Safe DC)



For detailed Ethernet instructions: https://www.solaredge.com/sites/default/files/se_three_phase_inverters_with_setapp_installation_guide.pdf#page=53&zoom=auto,-186,595

**Comment by chrisfirefox** [comment]
Thanks for the suggestion scheric! I already called my installer to do that for me - I've got a troubled history with electricity :). But the cable is installed and I'm confident that this will finally allow me real time access to the production data of my inverter.

**Comment by rogerthn2019** [comment]
> Thanks for the suggestion scheric! I already called my installer to do that for me - I've got a troubled history with electricity :). But the cable is installed and I'm confident that this will finally allow me real time access to the production data of my inverter.

Looks like I need to get my LAN through the wall :-(
Let's know if you are successful :-)

**Comment by rogerthn2019** [comment]
New firmware 4.7.26 today and now I get ERR_CONNECTION_TIMED_OUT on
`hXXp://<Inverter WiFi IP>`
nmap say:
`PORT     STATE    SERVICE`
`80/tcp   filtered http`
`8080/tcp filtered http-proxy`
Any comments on this?

**Comment by scheric** [comment]
@rogerthn2019 Are you on wifi or Ethernet.

My single phase is on Ethernet with 4.6.27
#### nmap -A 192.168.178.29
```
Not shown: 998 closed ports
PORT     STATE SERVICE VERSION
80/tcp   open  http    Werkzeug httpd 0.11.3 (Python 3.4.3)
|_http-title: Inverter SetApp

8080/tcp open  http    Werkzeug httpd 0.11.3 (Python 3.4.3)
|_http-title: Site doesn't have a title (text/html; charset=utf-8).

MAC Address: xx:xx:xx:xx:xx:xx (Ieee Registration Authority)
Device type: general purpose
Running: Linux 3.X|4.X
OS CPE: cpe:/o:linux:linux_kernel:3 cpe:/o:linux:linux_kernel:4
OS details: Linux 3.2 - 4.9
Network Distance: 1 hop
```

**Comment by rogerthn2019** [comment]
> @rogerthn2019 Are you on wifi or Ethernet.
> 
WiFi as advised by the, less skilled installer/seller :-(
and the inverter is outdoor :-( :-(

**Comment by bosgnoom** [comment]
Hello,
I've got the same issue as @rogerthn2019 .

Firmware 4.7.26, using a wired Ethernet connection.

**Comment by rogerthn2019** [comment]
> Hello,
> I've got the same issue as @rogerthn2019 .
> 
> Firmware 4.7.26, using a wired Ethernet connection.

Thanks @bosgnoom , good to know!
The inverter is responding to ping but no reply on port 80 or 8080 according to Wireshark
Maybe I need to have a look at https://github.com/jbuehl/solaredge or something similar and try to capture some traffic
